File: spa/src/app/models/DTOs/RoleUpdatePermissionsDto.ts
import { Role } from '../Entities/Role';
import { UpdateTypes } from '../Enums/UpdateTypes.enum';

export class RoleUpdatePermissionsDto {
    constructor(
        public role: Role,
        // public permissions: Permission[]
        public permissionIds: number[],
        public updateType: UpdateTypes
    ) { }
}

File: api/Engenharia.Domain/DTOs/RoleUpdatePermissionsDto.cs
using Engenharia.Domain.Enums;
using System.Collections.Generic;

namespace Engenharia.Domain.DTOs
{
    public class RoleUpdatePermissionsDto
    {
        public RoleDto Role { get; set; }
        //public List<PermissionDto> Permissions { get; set; }
        public List<int> PermissionIds { get; set; }
        public UpdateTypes UpdateType { get; set; }
    }

    
}
